Privatize AC_Avoidance Logging

hendjoshsr71 opened the pull request:
This moves avoidance logging functions out of AP_Logger and into AC_Avoidance. Also adds const to where it applies to the logging functions.

Note the `Write_SimpleAvoidance()` function is not called anywhere that I could find....??

**Before PR**
![master_OABR](https://user-images.githubusercontent.com/69225461/116647119-ceaabc00-a947-11eb-9e45-ff2ead5bf36a.png)
![master_OADJ](https://user-images.githubusercontent.com/69225461/116647122-cfdbe900-a947-11eb-9a60-1b2d1e5f2953.png)

**After PR**
![after_pr_OABR](https://user-images.githubusercontent.com/69225461/116647180-eaae5d80-a947-11eb-8817-5fcc37774c86.png)
![after_pr_OADJ](https://user-images.githubusercontent.com/69225461/116647183-ec782100-a947-11eb-899b-bdf738999768.png)

rishabsingh3003 commented:
Hi.
Yes, you are right, I added simple avoidance logging a while back, but then I removed it when we did the avoidance 3D conversion overhaul. I have been meaning to add it back, that's why I didn't remove the supporting functions. Basically, it needs a couple of new fields. I'll do it sometime soon

hendjoshsr71 commented:
> Hi.
> Yes, you are right, I added simple avoidance logging a while back, but then I removed it when we did the avoidance 3D conversion overhaul. I have been meaning to add it back, that's why I didn't remove the supporting functions. Basically, it needs a couple of new fields. I'll do it sometime soon

Ahh makes sense! Figured it might be something like that. Thanks for the clarification

peterbarker commented on libraries/AC_Avoidance/AP_OABendyRuler.cpp:
```diff
@@ -306,7 +306,7 @@ bool AP_OABendyRuler::search_vertical_path(const Location& current_loc, const Lo
                         _current_lookahead = MIN(_lookahead, _current_lookahead * 1.1f);
                     
-                        AP::logger().Write_OABendyRuler((uint8_t)OABendyType::OA_BENDY_VERTICAL, active, bearing_to_dest, pitch_delta, false, margin, destination, destination_new);
+                        Write_OABendyRuler((uint8_t)OABendyType::OA_BENDY_VERTICAL, active, bearing_to_dest, pitch_delta, false, margin, destination, destination_new);
```
This shouldn't be casting here, rather casting it in the function.

Pre-existing, so *shrug*.

peterbarker reviewed (approved):
LGTM

peterbarker commented:
Merged, thanks!
